lib/ReactViewModels/SearchState.ts

```diff
@@ -46,28 +42,36 @@ export default class SearchState {
   private _locationSearchDisposer: IReactionDisposer;
   private _unifiedSearchDisposer: IReactionDisposer;
 
+  private readonly terria: Terria;
+
   constructor(options: SearchStateOptions) {
     makeObservable(this);
-    this.catalogSearchProvider =
-      options.catalogSearchProvider ||
-      new CatalogSearchProvider({ terria: options.terria });
-    this.locationSearchProviders = options.locationSearchProviders || [];
+
+    this.terria = options.terria;
+
+    runInAction(() => {
+      this.terria.searchBarModel.catalogSearchProvider =
+        options.catalogSearchProvider ||
+        new CatalogSearchProvider("catalog-search-provider", options.terria);
+    });
+
+    const self = this;
 
     this._catalogSearchDisposer = reaction(
-      () => this.catalogSearchText,
+      () => self.catalogSearchText,
```

Comment:
I don't believe this is needed since arrow functions use the enclosing `this` value